agenthub/codeact_agent/codeact_agent.py

```diff
@@ -203,9 +203,9 @@ def step(self, state: State) -> Action:
         }
         try:
             response = self.llm.completion(**params)
-        except Exception:
+        except Exception as e:
             return AgentFinishAction(
-                thought='Agent encountered an error while processing the last action. Please try again.'
+                thought=f'Agent encountered an error while processing the last action.\nError: {str(e)}\nPlease try again.'
```

Comment:
I think this error from liteLLM can look very raw by default. IIRC we have intently removed displaying such raw errors from the UI in the past. Starting from here: https://github.com/All-Hands-AI/OpenHands/pull/1296

Instead we replaced them with some text for the user to know what happened, but in a user-friendly way. e.g. replace the original exception with a Malformed...SomethingException. This issue tracks what else or how else we can inform the user better: https://github.com/All-Hands-AI/OpenHands/issues/1451

Reply:
I've shortened the UI error message, at least it shouldn't have the "traceback" kinda code anymore.